soroban-env-host/benches/common/measure.rs

```diff
@@ -285,11 +299,49 @@ mod cpu {
     }
 }
 
-fn measure_costs_inner<HCM: HostCostMeasurement, F>(
+fn harness<HCM: HostCostMeasurement, R>(
+    host: &Host,
+    mem_tracker: &MemTracker,
+    cpu_insn_counter: &mut cpu::InstructionCounter,
+    alloc_group_token: &mut AllocationGroupToken,
+    iterations: u64,
+    runner: &mut R,
+    samples: Vec<<<HCM as HostCostMeasurement>::Runner as CostRunner>::SampleType>,
+) -> Measurement
+where
+    R: FnMut(&Host, Vec<<<HCM as HostCostMeasurement>::Runner as CostRunner>::SampleType>),
+{
+    // TODO get rid of dependency of inputs on samples
+    let sample = samples[0].clone();
+    // start the cpu and mem measurement
+    let start = Instant::now();
+    mem_tracker.0.store(0, Ordering::SeqCst);
+    let alloc_guard = alloc_group_token.enter();
+    host.as_budget().reset_unlimited();
```

Comment:
I think you can move the budget reset up before the memory-tracking activation, and move the call to Instant::now down below it (to right before the instruction counter begins)

Reply:
Fixed, that makes sense.